File: NapakalakiRuby/lib/monster.rb
#encoding: utf-8
#Autor: Dunia Rodríguez y Emilio Reguero

require_relative "prize.rb"
require_relative "bad_consequence.rb"

class Monster
  attr_reader :name, :combatLevel, :prize, :bc, :MAXTREASURES
  
  @@MAXTREASURES = 10 
  def initialize(name, level, prize, bc)
    @name = name
    @combatLevel = level
    @prize = prize
    @bc = bc
  end
  
  def getLevelsGained
    @prize.level
  end
  
  def getTreasuresGained
    @prize.treasures
  end
  
  def to_s
    "Monster: #{@name}, combatLevel: #{@combatLevel} \n #{@bc} \n #{@prize}"
  end
  
end

File: Napakalaki/src/napakalaki/Monster.java
/*
* PDOO Practicas - Napakalaki Java
* @author Emilio Reguero y Dunia Rodríguez 
*/
package napakalaki;


public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence bc;
    
    
    
    public Monster (String name, int combatLevel, BadConsequence bc, Prize prize) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.prize = prize;
        this.bc = bc;
    }
    
    String getName(){
        return name;
    }
    
    int getCombatLevel(){
        return combatLevel;
    }
    
    Prize getPrize(){
        return prize;
    }
    
    BadConsequence getBadConsequence(){
        return bc;
    }
    
    public int getlevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
   @Override
   public String toString(){
       return("\nName = " + name + ", CombatLevel: " + combatLevel + 
               ".\nPrize: " + prize + "\nBadConsequence: " + bc +".\n");
   }
}
